pkg/controller/oam/applicationconfiguration/render.go

```diff
@@ -102,6 +102,11 @@ func (r *components) Render(ctx context.Context, ac *v1alpha2.ApplicationConfigu
 				return nil, errors.Wrapf(err, errFmtRenderTrait, acc.ComponentName)
 			}
 
+			// Set metadata name for `Trait` if the metadata name is NOT set.
+			if t.GetName() == "" {
+				t.SetName(acc.ComponentName)
+			}
```

Comment:
What happens if two AppConfigs use the same Component? Won't we end up with multiple traits trying to use the same name?

Reply:
As discussed before, Component is an instance, one component should be in one appconfig 
 only.

Reply:
> As discussed before, Component is an instance, one component should be in one appconfig
only.

Not sure if we have admission to enforce this practice?

Reply:
What about use  componentName+<hash-suffix> to fix this case? I don't like appconfigName+componentName because it could be too long. @negz @resouer

Reply:
ref to admission controller issue https://github.com/crossplane/crossplane/issues/1509

Reply:
The current name is fine to me, I'm just suggesting we may need to include admission to enforce this.